[configure] Detect Dune 2.9, use right install options in this case. (#14844)
coq/coq: doc/sphinx/changes.rst

  by Emilio Jesus Gallego Arias, review by Vincent Laporte, Guillaume
  Melquiond, Enrico Tassi, and Théo Zimmerman).
- **Changed:**
  Coq's OCaml parts and tools [``coq-core``] are now built using Dune.
  The main user-facing change is that Dune >= 2.5 is now required to

>   build Coq, most recent Dune is usually recommended. This was a large and

Reviewer: ```suggestion
  build Coq.  We usually recommend using the most recent version of Dune. This was a large and
```

Can you say when would it be appropriate to use an older version of Dune?

Reviewer 2: There might be reasons why people want to use an older version, e.g. because they use their system package manager to install it. But the latest version is always recommended.

>   complex change, if you are packager you may find some minor differences

Reviewer: ```suggestion
  complex change.  If you are packager you may find some minor differences
```

>   if you were using a lot of custom optimizations. Note that in particular

Reviewer: ```suggestion
  if you were using a lot of custom optimizations. Note that, in particular,
```

  that configure option ``-datadir`` is not customizable anymore, and ``-bindir``
  has been removed in favor of ``$prefix/bin``. Moreover, the install procedure

>   will ignore ``-docdir`` and ``-etcdir`` if using Dune < 2.9.

Reviewer: ```suggestion
  will ignore ``-docdir`` and ``-etcdir`` for Dune < 2.9.
```

  to what ``coq_makefile`` does.
  (`#14258 <https://github.com/coq/coq/pull/14258>`_,
  by Emilio Jesus Gallego Arias).
- **Changed:**

>   The ``-docdir`` configure option now refers to root path for documentation,
>   if you would like to install Coq documentation in ``foo/coq``, use
>   ``-docdir foo``.

Reviewer: ```suggestion
  The ``-docdir`` configure option now refers to the root path for documentation.
  If you would like to install Coq documentation in ``foo/coq``, use
  ``-docdir foo``.
```
